### PR TITLE
[enhancement](nereids)eliminate repeat node if there is only 1 grouping set and no grouping scalar function

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes

convert
`SELECT k1, k2, SUM(k3) FROM d_table GROUP BY GROUPING SETS ((k1, k2));`
to
`SELECT k1, k2, SUM(k3) FROM d_table GROUP BY (k1, k2);`

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java [hunk 1]
```diff
@@ -116,6 +116,11 @@ private void checkGroupingSetsSize(LogicalRepeat<Plan> repeat) {
     }
 
     private LogicalAggregate<Plan> normalizeRepeat(LogicalRepeat<Plan> repeat) {
+        if (repeat.getGroupingSets().size() == 1 && ExpressionUtils
+                .collect(repeat.getOutputExpressions(), GroupingScalarFunction.class::isInstance).isEmpty()) {
+            return new LogicalAggregate<>(repeat.getGroupByExpressions(),
+                    repeat.getOutputExpressions(), Optional.empty(), repeat.child());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java:122]: we have three args ctor
```suggestion
            return new LogicalAggregate<>(repeat.getGroupByExpressions(),
                    repeat.getOutputExpressions(), repeat.child());
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java [hunk 2]
```diff
@@ -116,6 +116,11 @@ private void checkGroupingSetsSize(LogicalRepeat<Plan> repeat) {
     }
 
     private LogicalAggregate<Plan> normalizeRepeat(LogicalRepeat<Plan> repeat) {
+        if (repeat.getGroupingSets().size() == 1 && ExpressionUtils
+                .collect(repeat.getOutputExpressions(), GroupingScalarFunction.class::isInstance).isEmpty()) {
+            return new LogicalAggregate<>(repeat.getGroupByExpressions(),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java:121]: we move this code block out of normalizeRepeat. do it at the begin of this rule, aka at the very begin in `then`

File: regression-test/suites/nereids_rules_p0/grouping_sets/grouping_normalize_test.groovy [hunk 3]
```diff
@@ -39,4 +39,19 @@ suite("grouping_normalize_test"){
     SELECT  ROUND( SUM(pk  +  1)  -  3)  col_alias1,  MAX( DISTINCT  col_int_undef_signed  -  5)   AS col_alias2, pk  +  1  AS col_alias3
     FROM grouping_normalize_test  GROUP BY  GROUPING SETS ((col_int_undef_signed,col_int_undef_signed2,pk),()) order by 1,2,3;
     """
+
+    explain {
```
Comment (morrySnow) [inline regression-test/suites/nereids_rules_p0/grouping_sets/grouping_normalize_test.groovy:43]: also add a plan check test in [NormalizeRepeatTest.java](https://github.com/apache/doris/pull/35813/files#diff-96ad3ba35e43204328613a2967f79313c2caaf1e76d8bacd1a322768c542c578)

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by feiniaofeiafei — APPROVED** [review]

**Pull request merged**